client/app/scripts/components/node-details/node-details-image-status.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import find from 'lodash/find';
+import map from 'lodash/map';
+import { CircularProgress } from 'weaveworks-ui-components';
+
+import { getImagesForService } from '../../actions/app-actions';
+
+const topologyWhitelist = ['services'];
+
+function getNewImages(images, currentId) {
+  // Assume that the current image is always in the list of all available images.
+  // Should be a safe assumption...
+  const current = find(images, i => i.ID === currentId);
+  const timestamp = new Date(current.CreatedAt);
+  return find(images, i => timestamp < new Date(i.CreatedAt)) || [];
+}
+
+class NodeDetailsImageStatus extends React.PureComponent {
+  constructor(props, context) {
+    super(props, context);
+
+    this.handleServiceClick = this.handleServiceClick.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.props.serviceName) {
+      this.props.getImagesForService(this.props.params.orgId, this.props.serviceName);
+    }
+  }
+
+  handleServiceClick() {
+    this.props.router.push(`/flux/${this.props.params.orgId}/services/${encodeURIComponent(this.props.serviceName)}`);
+  }
+
+  render() {
+    const { errors, service, fetching, currentTopologyId } = this.props;
+
+    if (currentTopologyId && !topologyWhitelist.includes(currentTopologyId)) {
+      return null;
+    }
+
+    return (
+      <div className="node-details-content-section image-status">
+        <div className="node-details-content-section-header">
+          Container Image Status
+          {service &&
+            <div>
+              <a
+                onClick={this.handleServiceClick}
+                className="node-details-table-node-link">
+                  View in Deploy
+              </a>
+            </div>
+          }
+
+        </div>
+        {fetching && <CircularProgress />}
+        {!fetching && errors && <p>Error: {JSON.stringify(map(errors, 'message'))}</p>}
+        {!errors && !fetching && !service && 'No service images found'}
+        {!errors && !fetching && service &&
```

Comment:
These set of conditions looks like hard to go through (and more importantly, to immidiately see whether they are complementary and complete), maybe use some ternary ` ? : ` or subrendering helpers?